sources/packages/backend/libs/integrations/src/services/disbursement-schedule/e-cert-generation.service.ts

```diff
@@ -52,585 +28,242 @@ const DISBURSEMENT_SCHEDULES_UPDATE_CHUNK_SIZE = 1000;
-    // Step 2 - Execute the calculation to define the final value to be used for the e-Cert.
-    await this.calculateEffectiveValue(disbursements, entityManager);
-    // Step 3 - Calculate tuition remittance effective amount.
-    this.calculateTuitionRemittanceEffectiveAmount(disbursements);
-    // Step 4 - Calculate BC total grants after all others calculations are done.
-    //!This step relies on the effective value calculation (step 2).
-    await this.createBCTotalGrants(disbursements);
-    // Step 5 - Mark all disbursements as 'sent'.
-    const now = new Date();
-    disbursements.forEach((disbursement) => {
-      disbursement.disbursementScheduleStatus = DisbursementScheduleStatus.Sent;
-      disbursement.dateSent = now;
+    const eligibleApplications = await this.applicationRepo.find({
+      select: {
+        id: true,
+        currentAssessment: {
+          id: true,
+          disbursementSchedules: {
+            id: true,
+            disbursementDate: true,
+            tuitionRemittanceRequestedAmount: true,
+            coeStatus: true,
+            disbursementValues: {
+              id: true,
+              valueType: true,
+              valueCode: true,
+              valueAmount: true,
+              disbursedAmountSubtracted: true,
+            },
+            msfaaNumber: {
+              id: true,
+              dateSigned: true,
+              cancelledDate: true,
+            },
+          },
+          offering: {
+            id: true,
+            offeringIntensity: true,
+            actualTuitionCosts: true,
+            programRelatedCosts: true,
+          },
+        },
+        student: {
+          id: true,
+          sinValidation: {
+            id: true,
+            isValidSIN: true,
+          },
+          studentRestrictions: {
+            id: true,
+            restriction: {
+              id: true,
+              actionType: true,
+            },
+          },
+        },
+        programYear: {
+          id: true,
+          maxLifetimeBCLoanAmount: true,
+        },
+      },
+      relations: {
+        currentAssessment: {
+          disbursementSchedules: {
+            disbursementValues: true,
+            msfaaNumber: true,
+          },
+          offering: true,
+        },
+        student: {
+          sinValidation: true,
+          studentRestrictions: { restriction: true },
+        },
+        programYear: true,
+      },
+      where: {
+        applicationStatus: ApplicationStatus.Completed,
+        currentAssessment: {
+          disbursementSchedules: {
+            disbursementScheduleStatus: DisbursementScheduleStatus.Pending,
+            disbursementDate: LessThanOrEqual(disbursementMinDate),
+          },
+          offering: {
+            offeringIntensity,
+            studyEndDate: Raw(
+              (endDate) =>
+                `(${endDate} + ${this.configService.applicationArchiveDays}) >= CURRENT_DATE`,
```

Comment:
Question here. we could have added application archive days to study end date and used a similar comparison like line 119 right?

Reply:
If I am following correctly, line 119 uses a "constant"(disbursementMinDate) different than here where the `studyEndDate` would be different for every single record.
Please let me know if I answered accordingly.